scene/resources/animation.cpp

```diff
@@ -83,9 +83,11 @@ bool Animation::_set(const StringName &p_name, const Variant &p_value) {
 
 				const real_t *r = values.ptr();
 
-				tt->transforms.resize(vcount / 12);
+				int transform3d_size = (int)sizeof(Transform3D);
 
-				for (int i = 0; i < (vcount / 12); i++) {
+				tt->transforms.resize(vcount / transform3d_size);
+
+				for (int i = 0; i < (vcount / transform3d_size); i++) {
 					TKey<TransformKey> &tk = tt->transforms.write[i];
 					const real_t *ofs = &r[i * 12];
```

Comment:
I wonder if all the 12 constants should be replaced.

Reply:
Maybe we should add something like this: `(sizeof(Quaternion) / sizeof(real_t)) + (sizeof(Vector3) / sizeof(real_t)) + (sizeof(Vector3) / sizeof(real_t)) + 2`

Reply:
Are we replacing the 12 valued constants?

Reply:
Yes. For the readability, we can replace.
Example:
`int transform3d_var_count = 12`
or
`int transform3d_var_count = (sizeof(Quaternion) / sizeof(real_t)) + (sizeof(Vector3) / sizeof(real_t)) + (sizeof(Vector3) / sizeof(real_t)) + 2`

But this is not related to this issue, we need to push another commit/pr

Reply:
@requizm Transform3D doesn't contain Quaternion, it contains one Basis and one separate Vector3 for the origin.

Reply:
sizeof(Basis) is 9 floats vs sizeof(Quaternion) which is 4 floats.

Reply:
@requizm This code is quite confusing actually because `transform3d_var_count` is not the variable count of `Transform3D`, it is the variable count of `TransformKey`'s members plus `Key`'s `transition` and `time`.

Reply:
> transition

You're right. Bad naming.